### PR TITLE
Classifier: react to workflow version changes

### PR DESCRIPTION
**Pull request by eatyourgreens**
Replace the classifier's `workflowData` prop with `workflowSnapshot` and `workflowVersion` props. `workflowData` changes when eg. workflow completeness changes, which isn't very useful. This change restricts updates (and re-renders) to cases where `workflow.version` has explicitly changed.

Closes: #2749.

# Review Checklist

## General

- [x] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [x] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

File: packages/lib-classifier/src/components/Classifier/ClassifierContainer.js
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/ClassifierContainer.js:152]: The logic here needs to be inverted: render the classifier if loaded and if the snapshot is loaded. Otherwise show a loading message.

**Review by shaunanoordin — APPROVED** [review]
## PR Review

Note: this review is parallel & supplemental to @mcbouslog's review

This PR fixes 2749 by ensuring the "did the workflow change?" action only triggers when _the workflow version changes,_ (i.e. when its primary data is changed,) not when its derived values (e.g. completeness % ) change.

Note: "did the workflow change?" also correctly triggers when the workflow changes (e.g. when first loading the Classifier, or when going from WF1 to WF2.)

### Testing

Testing is limited as I haven't been able to fully replicate the "change in workflow.completeness" scenario on localhost.

In any case, here are the tested scenarios:

Scenario 1: standard Workflow load
- I can load Workflow X in the Classifier normally (view Subject and Task) if I...
  - ...start at the project home page and click on WF X
  - ...start at the project home page and click on Classify, then click WF X on "choose your workflow"
  - ...start at the /classify route and then click WF X on "choose your workflow"
  - ...go directly to the WF X URL, e.g. https://local.zooniverse.org:3000/projects/darkeshard/test-project-2021/classify/workflow/3580

Scenario 2: switching to a different Workflow
- If I start at WF X [(example)(https://local.zooniverse.org:3000/projects/darkeshard/test-project-2021/classify/workflow/3580), I can click on Classify to open the "choose your workflow" menu, then click on WF Y
- WF Y will load normally in the Classifier

Scenario 3: active Workflow editing
- As a Project Owner, if I have WF X open in browser tab 1 [(example)](https://local.zooniverse.org:3000/projects/darkeshard/test-project-2021/classify/workflow/3580) and the Project Builder for WF X open in tab 2... [(example)](https://master.pfe-preview.zooniverse.org/lab/1873/workflows/3580)
  - ...then, if I edit WF X in tab 2, when I switch focus back to the tab 1, the changes will be reflected automatically. (i.e. swr will fetch the latest WF X, realise it has a different version number, and re-render the Classifier accordingly)
  - On a WF update, **the active Subject does not change.**
  - On a WF update, **all active annotations (i.e. the Classification) will reset.**
  - This should be the _only scenario_ where a WF update can happen **in mid classification**

❗ Scenario 4: **Not tested**
- As a user, say I'm classifying Subject 1234 on WF X on tab 1. [(example)](https://local.zooniverse.org:3000/projects/darkeshard/test-project-2021/classify/workflow/3580)
- Then, say I switch to tab 2 and _somehow increase the completion of WF X..._
  - _(Maybe by classifying as a different user on a different browser and cranking the retirement required counter down to 1?)_
- ...when I switch back to tab 1, **my classifications should not disappear.**

Tested on localhost with macOS + Chrome using a [random test project.](https://local.zooniverse.org:3000/projects/darkeshard/test-project-2021/classify/workflow/3580)

### Status

LGTM 👍 Waiting on Mark's review before this should be considered safe to merge.

**Review by mcbouslog — APPROVED** [review]
Code changes LGTM, tested on TESS, HSM NHS, Davy Notebooks, and a test project (with multiple workflows, switched between) without any issues noted.

**Pull request merged**
